packages/core-flows/src/handlers/sales-channel/create-sales-channels.ts

```diff
@@ -0,0 +1,35 @@
+import { createStep, StepResponse } from "@medusajs/workflows-sdk"
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import {
+  CreateSalesChannelDTO,
+  ISalesChannelModuleService,
+  SalesChannelDTO,
+} from "@medusajs/types"
+
+export const createSalesChannelsStep = createStep<
+  CreateSalesChannelDTO[],
+  SalesChannelDTO[],
+  string[]
+>(
+  "create-sales-channels",
+  async (data: CreateSalesChannelDTO[], { container }) => {
+    const salesChannelService: ISalesChannelModuleService = container.resolve(
+      ModuleRegistrationName.SALES_CHANNEL
+    )
+
+    const createdChannels = await salesChannelService!.create(data)
+    return new StepResponse(
+      createdChannels,
+      createdChannels.map((channel) => channel.id)
+    )
+  },
+  async (createdChannelsIds, { container }) => {
+    const salesChannelService: ISalesChannelModuleService = container.resolve(
+      ModuleRegistrationName.SALES_CHANNEL
+    )
+
+    if (createdChannelsIds?.length) {
```

Comment:
todo: early return here
```suggestion
    if (!createdChannelsIds?.length) {
      return
    }
    
    const salesChannelService: ISalesChannelModuleService = container.resolve(
      ModuleRegistrationName.SALES_CHANNEL
    )
    await salesChannelService!.delete(createdChannelsIds)
```